planner/core/point_get_plan.go

```diff
@@ -767,6 +767,20 @@ func tryPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt) *PointGetP
 		return p
 	}
 
+	check = check && ctx.GetSessionVars().ConnectionID > 0 && len(tbl.Indices) > 0
+	var latestIndexes map[int64]*model.IndexInfo
+	var err error
+	loadLatestIndexes := func() bool {
+		latestIndexes, err = getLatestIndexInfo(ctx, tbl.ID, 0)
+		if err != nil {
+			return false
+		}
+		if latestIndexes == nil {
+			check = false
+		}
+		return true
+	}
```

Comment:
We could abstract it in a function and return the result assigning to `check` varible, as well as the `error` value.